Quick_Select/Quick_Select.cpp

```diff
@@ -0,0 +1,112 @@
+/*
+    Quick Select Algorithm is used for extracting the kth smallest
+    or kth largest element from the array without actually sorting
+    the array. It uses the priciple of Quick Sort Algorithm, where
+    we partition the array and select a pivot. The elements lesser
+    than the pivot lie to its left, and the elements greater than 
+    the pivot lie to its right. We can observe that in such scenario
+    the pivot element is actually in its correct place and hence is
+    the kth smallest element. The best and average time complexity of
+    Quick Select is O(n). The worst case complexity is O(n^2).
+*/
+
+#include <bits/stdc++.h>
+using namespace std;
+
+// Function to partition the array
+int partition(int arr[], int left, int right, int pivotIndex) 
+{  
+    // Select pivot element from the pivot index
+    int pivotElement = arr[pivotIndex];
+
+    // Move pivot to the right
+    swap(arr[pivotIndex], arr[right]);
+
+    // Start moving from left of the array
+    int pIndex = left;
+    
+    /*
+        Whenever we find an element lesser than the pivot,
+        move it towards the left of the array, this way the
+        elements greater than the pivot accumulate towards 
+        the right of the array, and finally we will swap the
+        pIndex and pivot, so that the pivot lies at its
+        correct sorted place.
+    */ 
+    for (int i = left; i < right; i++) 
+    {
+        if (arr[i] <= pivotElement) 
+        {
+            swap( arr[i], arr[pIndex] );
```

Comment:
Please remove space before arr[i] and after arr[pIndex].

Reply:
Done